system/BaseModel.php

```diff
@@ -780,6 +781,67 @@ public function getInsertID()
         return is_numeric($this->insertID) ? (int) $this->insertID : $this->insertID;
     }
 
+    /**
+     * Validates that the primary key values are valid for update/delete/insert operations.
+     * Throws exception if invalid.
+     *
+     * @param int|list<int|string>|RawSql|string $id
+     * @param bool                               $allowArray Whether to allow array of IDs (true for update/delete, false for insert)
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function validateID($id, bool $allowArray = true): void
```

Comment:
Basically, this allows `mixed $id` but after passing it to this method, the `$id` is now narrowed to the indicated types in param. We can precisely say this by:

```suggestion
     * @param bool                               $allowArray Whether to allow array of IDs (true for update/delete, false for insert)
     * @phpstan-assert int|list<int|string>|RawSql|string $id
     * @throws InvalidArgumentException
     */
    protected function validateID(mixed $id, bool $allowArray = true): void
```